### PR TITLE
[feature][doc] Generate client config docs from source code

### PR DESCRIPTION
**Pull request by SignorMercurio**
Master issue: #14184

### Motivation

Please checkout the master issue.

### Modifications

Add some docs annotations to `Client`, `Producer`, `Consumer` and `Reader` with the data sourcing from the configuration table from [Pulsar Java Client Docs](https://pulsar.apache.org/docs/next/client-libraries-java), so that their config docs can be auto-generated. The code to auto-generate these are also added in this PR, while the scripts are added in apache/pulsar-site#176.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)

This PR is about docs generation only.

- [] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)
- [x] `doc`

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@urfreespace 
Could you please review this PR from the technical perspective? 
Thank you! 😊

File: pulsar-client/pom.xml [hunk 1]
```diff
@@ -182,6 +182,12 @@
       <artifactId>awaitility</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.pulsar</groupId>
+      <artifactId>pulsar-broker-common</artifactId>
+      <version>2.11.0-SNAPSHOT</version>
```
Comment (urfreespace) [inline pulsar-client/pom.xml:188]: I think we should use variables here, but not a specific version `2.11.0-SNAPSHOT`

**Review by urfreespace — CHANGES_REQUESTED** [review]

File: pulsar-client/pom.xml (hunk 1 above)
Comment (SignorMercurio) [reply pulsar-client/pom.xml:188]: Do you mean `${project.parent.version}`?

Comment (urfreespace) [reply pulsar-client/pom.xml]: OK, I see now

**Review by urfreespace — APPROVED** [review]

**Comment by Anonymitaet** [comment]
@Technoboy- Could you please review this PR from the technical perspective?
Thank you! 😊

**Review by Technoboy- — APPROVED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Review by RobertIndie — APPROVED** [review]

**Comment by Anonymitaet** [comment]
@SignorMercurio can you resolve conflicts? Then we can merge this

**Comment by SignorMercurio** [comment]
@Anonymitaet I've updated with the master branch. I think there's currently no conflict.

**Comment by nicoloboschi** [comment]
@SignorMercurio we must not add new dependencies to pulsar-client, especially pulsar-broker internal modules.
we can move [pulsar-broker-common/src/main/java/org/apache/pulsar/broker/BaseGenerateDocumentation.java](https://github.com/apache/pulsar/pull/17198/files#diff-6d4deed273c647ddf14385ae9f883834e94a2d411268be4ebec7b026b064a193) to the `pulsar-common` module in order to remove the need of `pulsar-broker-common` dependency in the pulsar-client

**Pull request merged**